src/main.py

```diff
@@ -495,75 +475,68 @@ def silence_listener(throwaway_frames):
 	inp.close()
 
 
-def loop():
-	global player
+trigger_thread = None
+def trigger_callback(trigger_name):
+	global trigger_thread
 
-	while True:
-		record_audio = False
+	triggers.disable()
+
+	trigger_thread = threading.Thread(target=trigger_process, args=(trigger_name,))
+	trigger_thread.setDaemon(True)
+	trigger_thread.start()
 
-		# Enable reading microphone raw data
-		inp = alsaaudio.PCM(alsaaudio.PCM_CAPTURE, alsaaudio.PCM_NORMAL, config['sound']['input_device'])
-		inp.setchannels(1)
-		inp.setrate(16000)
-		inp.setformat(alsaaudio.PCM_FORMAT_S16_LE)
-		inp.setperiodsize(1024)
 
-		while not record_audio:
-			time.sleep(.1)
+def trigger_process(trigger_name):
+	global player
+
+	trigger = triggers.triggers[trigger_name]
 
-			triggered = False
-			# Process microphone audio via PocketSphinx, listening for trigger word
-			while not triggered:
-				# Read from microphone
-				_, buf = inp.read()
-				# Detect if keyword/trigger word was said
-				decoder.process_raw(buf, False, False)
+	if player.is_playing():
+		player.stop()
 
-				triggered_by_platform = platform.should_record()
-				triggered_by_voice = decoder.hyp() is not None
-				triggered = triggered_by_platform or triggered_by_voice
+	if trigger.voice_confirm:
+		player.play_speech(resources_path + 'alexayes.mp3')
 
-			if player.is_playing():
-				player.stop()
+	# clean up the temp directory
+	if not debug:
+		for some_file in os.listdir(tmp_path):
+			file_path = os.path.join(tmp_path, some_file)
+			try:
+				if os.path.isfile(file_path):
+					os.remove(file_path)
+			except Exception as exp: # pylint: disable=broad-except
+				print(exp)
```

Comment:
I'd put the cleanup after everything has been done as to not slow down the response process. Maybe just before triggers.enable()?